src/lib/utils/__tests__/getReversedHighlight-test.ts

```diff
@@ -0,0 +1,39 @@
+import getReversedHighlight from '../getReversedHighlight';
+
+describe('getReversedHighlight', () => {
+  describe('reverse highlighted parts in a string', () => {
+    test('with an empty string', () => {
+      expect(getReversedHighlight('')).toEqual('');
+    });
+
+    test('with space only string', () => {
+      expect(getReversedHighlight('         ')).toEqual('         ');
+    });
+
+    test('with one match', () => {
+      expect(getReversedHighlight('this is a <mark>match</mark>')).toEqual(
+        '<mark>this</mark> <mark>is</mark> <mark>a</mark> match'
```

Comment:
The algorithm in Autocomplete outputs the following format, which results in less markup:

```html
<mark>cell </mark>phone<mark> accessories disc </mark>ch<mark>argers opp</mark>
```

Not sure which is best.

Reply:
I wonder if this could be better:

```
___pre____post___pre___post___

swap tags

___post____pre___post___pre___

add pre in the beginning and post in the end

pre___post____pre___post___pre___post

optional: remove empty tags in beginning and end, if like this

prepost____pre___post___prepost

to this

____pre___post___

```

Reply:
> The algorithm in Autocomplete outputs the following format, which results in less markup:
> 
> ```
> <mark>cell </mark>phone<mark> accessories disc </mark>ch<mark>argers opp</mark>
> ```
> 
> Not sure which is best.

The output is like that because I've followed the actual `highlight` schema which avoid adding characters other than alphanumerical in `<mark>`. If it's not necessary, it's indeed better this way and would reduce the complexity of the function